path_roles.go

```diff
@@ -208,6 +218,53 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		role.PermanentlyDelete = false
 	}
 
+	// update and validate SignInAudience if provided
+	if signInAudience, ok := d.GetOk("sign_in_audience"); ok {
+		signInAudienceValue, ok := signInAudience.(string)
+		if !ok {
+			return logical.ErrorResponse("Invalid type for sign_in_audience field. Expected string."), nil
+		}
+
+		validSignInAudiences := []string{"AzureADMyOrg", "AzureADMultipleOrgs", "AzureADandPersonalMicrosoftAccount", "PersonalMicrosoftAccount"}
+		valid := false
+		for _, validValue := range validSignInAudiences {
+			if signInAudienceValue == validValue {
+				valid = true
+				break
+			}
+		}
+		if !valid {
+			validValuesString := strings.Join(validSignInAudiences, ", ")
+			return logical.ErrorResponse("Invalid value for sign_in_audience field. Valid values are: %s", validValuesString), nil
+		}
+		role.SignInAudience = signInAudienceValue
+	}
+
+	// update and validate Tags if provided
+	if tags, ok := d.GetOk("tags"); ok {
+		if tagsList, ok := tags.([]string); ok {
+			uniqueTags := make(map[string]struct{})
+			for _, tag := range tagsList {
+				// Check individual tag size
+				if len(tag) < 1 || len(tag) > 256 {
+					return logical.ErrorResponse("individual tag size must be between 1 and 256 characters (inclusive)"), nil
+				}
+				// Check for whitespaces
+				if strings.Contains(tag, " ") {
+					return logical.ErrorResponse("whitespaces are not allowed in tags"), nil
+				}
+				// Check for duplicates
+				if _, exists := uniqueTags[tag]; exists {
+					return logical.ErrorResponse("duplicate tags are not allowed"), nil
+				}
```

Comment:
I'm wondering if it'd be a little more forgiving and user-friendly to trim+deduplicate the tags for users? Some Vault APIs do this using [`RemoveDuplicates`](https://github.com/hashicorp/go-secure-stdlib/blob/main/strutil/strutil.go#L235).

Reply:
Nice! Thank you. I added this